signwriting_transcription/pose_to_signwriting/data/prepare_poses.py

```diff
@@ -93,33 +106,80 @@ def process(args, pumping: bool = False):
             n_frames = backup[i % len(backup)]["n_frames"]
             trg = backup[i % len(backup)]["trg"]
             src = backup[i % len(backup)]["src"]
+            split = backup[i % len(backup)]["split"]
             all_data.append({
                 "id": f'{utt_id}({i})',  # unique id
                 "src": src,
                 "n_frames": n_frames,
-                "trg": trg
+                "trg": trg,
+                "split": split
             })
+    return all_data
+
+
+def process(args):
+    # pylint: disable=too-many-locals
+    dataset_root, data_root, name, tokenizer_type, set_split, pumping = (
+        args.dataset_root, args.data_root, args.dataset_name, args.tokenizer_type,
+        args.set_split, args.pumping)
+    cur_root = Path(data_root).absolute()
+    cur_root = cur_root / name
+
+    # dir for filterbank (shared across splits)
+    feature_root = cur_root / f"fbank{N_MEL_FILTERS}"
+    feature_root.mkdir(parents=True, exist_ok=True)
+
+    # Extract features
+    print(f"Create pose {name} dataset.")
+
+    print("Fetching train split ...")
+    dataset = load_dataset(dataset_root, set_split)
+
+    print("Extracting pose features ...")
+    for instance in dataset:
+        utt_id = instance[0]
+        extract_to_matrix(instance[1], feature_root / f'{utt_id}.npy', overwrite=False)
+
+    # Pack features into ZIP
+    print("ZIPing features...")
+    create_zip(feature_root, feature_root.with_suffix(".zip"))
+
+    if not set_split:
```

Comment:
same comment about `set_split`, i don't think this works, and maybe it should just be mandatory